### PR TITLE
Glb texture coordinate fix

### PR DESCRIPTION
**Pull request by slimbuck**
We address here the issue with glTF having texture coordinates with top-left origin while the engine has bottom-right origin. We previously dealt with this by _not_ flipping glTF embedded textures (where all others were being flipped). The change in this PR is to flip glTF texture coordinates instead and also flip textures, thus bringing them inline with the rest of the engine.

 The following changes are made:
- flip glTF texture V coordinates at load time
- flip glTF textures to match the rest of the engine
- fix for cubemap faces under ImageBitmap
- reenable ImageBitmap
- also simplify glTF vertex buffer creation function

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/resources/parser/glb-parser.js [hunk 1]
```diff
@@ -182,22 +185,81 @@ var generateNormals = function (sourceDesc, vertexDesc, positions, numVertices,
...
         offset: 0,
         stride: 12,
-        count: numVertices
+        count: numVertices,
+        components: 3,
+        type: TYPE_FLOAT32
+    };
+};
+
+var flipTexCoordVs = function (vertexBuffer) {
```
Comment (mvaligursky) [inline src/resources/parser/glb-parser.js:199]: t would be probably simpler and faster to pass semantic as a parameter to this function, and call it twice (uv0, uv1) then doing array in array loop when copying

File: src/resources/parser/glb-parser.js [hunk 2]
```diff
@@ -182,22 +185,81 @@ var generateNormals = function (sourceDesc, vertexDesc, positions, numVertices,
...
+                case TYPE_FLOAT32:
+                    floatOffsets.push({ offset: element.offset / 4 + 1, stride: element.stride / 4 });
+                    break;
+                case TYPE_UINT16:
+                    shortOffsets.push({ offset: element.offset / 2 + 1, stride: element.stride / 2 });
+                    break;
+                case TYPE_UINT8:
+                    byteOffsets.push({ offset: element.offset + 1, stride: element.stride });
+                    break;
+            }
```
Comment (mvaligursky) [inline src/resources/parser/glb-parser.js:219]: maybe print error message in default:

Comment (slimbuck) [reply src/resources/parser/glb-parser.js:219]: The glTF spec only supports these 3 data types (it's not an implementation choice on our part) don't think a message is necessary?

File: src/resources/parser/glb-parser.js (hunk 1 above)
Comment (slimbuck) [reply src/resources/parser/glb-parser.js:199]: Not sure I follow. This function could be called twice, but as far as I can see that just moves code from here to the caller. Perhaps I misunderstand?

File: src/resources/parser/glb-parser.js (hunk 2 above)
Comment (mvaligursky) [reply src/resources/parser/glb-parser.js:217]: perfect in that case, not needed.

File: src/resources/parser/glb-parser.js (hunk 1 above)
Comment (mvaligursky) [reply src/resources/parser/glb-parser.js:197]: I take it back. I was looking at this

```
var flip = function (offsets, type, one) {
        var typedArray = new type(vertexBuffer.storage);
        for (i = 0; i < offsets.length; ++i) {
            var index = offsets[i].offset;
            var stride = offsets[i].stride;
            for (j = 0; j < vertexBuffer.numVertices; ++j) {
                typedArray[index] = one - typedArray[index];
                index += stride;
            }
        }
    };
```
and thought inside look is the one that loops over offsets. It's the other way, so all good.

Comment (slimbuck) [reply src/resources/parser/glb-parser.js:197]: kk

**Pull request merged**